player/command.c

```diff
@@ -2404,6 +2494,23 @@ static int mp_property_vsync_jitter(void *ctx, struct m_property *prop,
     return m_property_double_ro(action, arg, stddev);
 }
 
+static int mp_property_display_resolution(void *ctx, struct m_property *prop,
+                                          int action, void *arg)
+{
+    MPContext *mpctx = ctx;
+    struct vo *vo = mpctx->video_out;
+    if (!vo)
+        return M_PROPERTY_UNAVAILABLE;
+    struct mp_rect res;
+    if (vo_control(vo, VOCTRL_GET_DISPLAY_RES, &res) <= 0)
+        return M_PROPERTY_UNAVAILABLE;
+    if (strcmp(prop->name, "display-width") == 0) {
+        return m_property_int_ro(action, arg, mp_rect_w(res));
+    } else {
+        return m_property_int_ro(action, arg, mp_rect_h(res));
+    }
+}
```

Comment:
If you're going to unconditionally turn a 4-values rectangle into a two-values size, then the API has a redundancy. It should instead pass only two values: width and height.

Reply:
mp_rect was chosen mainly since it's convenient. I guess I could return a int[] or something here instead.

Reply:
Should be okay now. Just mac needs updating later.